runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -752,11 +752,12 @@ static size_t rdma_threshold = DEFAULT_RDMA_THRESHOLD;
 #define UI64_TO_VP(x)     ((void*) (intptr_t) (x))
 
 //
-// Maximum number of PUTs in a chained transaction list.  This number
-// was determined empirically (on XC/Aries).  Doing more than this at
-// once didn't seem to improve performance.
+// Maximum number of PUTs/AMOs in a chained transaction list.  This
+// number was determined empirically (on XC/Aries).  Doing more than
+// this at once didn't seem to improve performance.
 //
 #define MAX_CHAINED_PUT_LEN 64
+#define MAX_CHAINED_AMO_LEN 64
```

Comment:
FYI I did tuning on this number and did not just copy the put number. I see max performance with anything above 32, and saw degraded performance with 128 or higher.